### PR TITLE
Fix dispose pattern

### PR DESCRIPTION
**Pull request by sebastienros**

### CODE REVIEW

File: src/YesSql.Core/Session.cs [hunk 1]
```diff
@@ -900,7 +908,15 @@ public async ValueTask DisposeAsync()
 
             _disposed = true;
 
-            await CommitOrRollbackTransactionAsync();
+            try
```
Comment (hishamco) [inline src/YesSql.Core/Session.cs:911]: We can create an inner function instead of writing code twice

File: src/YesSql.Core/Session.cs [hunk 2]
```diff
@@ -900,7 +908,15 @@ public async ValueTask DisposeAsync()
...
+            {
+                await CommitOrRollbackTransactionAsync();
+            }
+            catch
+            {
+                _transaction = null;
```
Comment (hishamco) [inline src/YesSql.Core/Session.cs:917]: I think this should go into a `finally` block for clean up

**Review by hishamco — CHANGES_REQUESTED** [review]

File: src/YesSql.Core/Session.cs (hunk 2 above)
Comment (sebastienros) [reply src/YesSql.Core/Session.cs:917]: No because the standard path does it if it doesn't fail

File: src/YesSql.Core/Session.cs (hunk 1 above)
Comment (sebastienros) [reply src/YesSql.Core/Session.cs:911]: No, one is async the other on is not

**Pull request merged**